crates/ruff_python_formatter/src/pattern/pattern_match_sequence.rs

```diff
@@ -51,3 +43,54 @@ impl FormatNodeRule<PatternMatchSequence> for FormatPatternMatchSequence {
         }
     }
 }
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub(crate) enum SequenceType {
+    /// A list literal, e.g., `[1, 2, 3]`.
+    List,
+    /// A parenthesized tuple literal, e.g., `(1, 2, 3)`.
+    Tuple,
+    /// A tuple literal without parentheses, e.g., `1, 2, 3`.
+    TupleNoParens,
+}
+
+impl SequenceType {
+    pub(crate) fn from_pattern(pattern: &PatternMatchSequence, source: &str) -> SequenceType {
+        if source[pattern.range()].starts_with('[') {
+            SequenceType::List
+        } else if source[pattern.range()].starts_with('(') {
+            let Some(elt) = pattern.patterns.first() else {
+                return SequenceType::Tuple;
+            };
+
```

Comment:
Can you give an example of what this means? From what I understand, there are no patterns but it does start with an opening parentheses i.e., is this an empty tuple `()`?